### PR TITLE
Switch from invididual invariant arguments to an "Invariants" class

### PR DESCRIPTION
**Pull request by maceonthompson**
Re: advice from Fabrice

### CODE REVIEW

File: src/main/java/com/google/sps/data/Invariants.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  public float getMinCredits(){
+    return this.minCredits;
+  }
+
+  public float getMaxCredits(){
+    return this.maxCredits;
+  }
+}
```
Comment (achillesecos) [inline src/main/java/com/google/sps/data/Invariants.java]: New line here.

Comment (maceonthompson) [reply src/main/java/com/google/sps/data/Invariants.java]: Resolved!

**Review by achillesecos — COMMENTED** [review]
Just wondering, are you going to include the other 3 invariants here? 

(From Design Doc)
- 1.  No courses will overlap in a schedule
- 2.  The total number of credits contained in a schedule must be within the min/max bound of credits (specified by the user)
- 3.  If a course is marked as required, it must be in the schedule.
- 4. There can’t be multiple sections of the same course in the same schedule

Seems like you check invariant 4 in Section.java, will you check for the other invariants here?

**Comment by maceonthompson** [comment]
> Just wondering, are you going to include the other 3 invariants here?
> 
> (From Design Doc)
> 
> * 1. No courses will overlap in a schedule
> * 1. The total number of credits contained in a schedule must be within the min/max bound of credits (specified by the user)
> * 1. If a course is marked as required, it must be in the schedule.
> * 1. There can’t be multiple sections of the same course in the same schedule
> 
> Seems like you check invariant 4 in Section.java, will you check for the other invariants here?

No. This serves as a capsule for invariants that are changeable by the user (which for right now is the min/max number of credits). So it's more of a wrapper for actual data as opposed to different logical implementations. The ideas of no course overlap, same section overlap, and required courses are more changes in the logic of the `generateSchedules()` function in the scheduler class.

**Comment by achillesecos** [comment]
Ok sounds good. I understand what you mean now.

**Review by achillesecos — APPROVED** [review]

**Review by kristen-law — APPROVED** [review]
LGTM

**Pull request merged**